web/src/settings_profile_fields.js

```diff
@@ -318,20 +321,39 @@ function open_custom_profile_field_form_modal() {
     });
 }
 
+function disable_submit_btn_if_empty_choices() {
+    const $choice_text_rows = $("#edit-custom-profile-field-form-modal .choice-row").find(
+        ".modal_text_input",
+    );
+    let non_empty_choice_present = false;
+    for (const text_row of $choice_text_rows) {
+        if ($(text_row).val() !== "") {
+            non_empty_choice_present = true;
+            break;
+        }
+    }
+    $("#edit-custom-profile-field-form-modal .dialog_submit_button").prop(
+        "disabled",
+        !non_empty_choice_present,
+    );
+}
+
 function add_choice_row(e) {
-    if ($(e.target).parent().next().hasClass("choice-row")) {
+    disable_submit_btn_if_empty_choices();
+    const $curr_choice_row = $(e.target).parent();
+    if ($curr_choice_row.next().hasClass("choice-row")) {
         return;
     }
+    // Display delete button on newly created choice
```

Comment:
It is for the empty choice field after we type some text in it, we don't display the delete button of the very last bottom choice field(the extra empty one).

I'll change it  to `Display delete button on text input in the extra empty choice field`

Reply:
"extra empty choice" still seems not clear enough without manually testing the UI. Probably being some more verbose and having something like below would be better -
`Display delete buttons for all existing choices before creating the new row, which will not have the delete button so that there is at least one option present.`